lib/Transforms/Utils/Local.cc

```diff
@@ -43,33 +43,36 @@ namespace seahorn
     
     for (BasicBlock &BB : F)
     {
-      if (region.count (&BB) <= 0) 
+      if (region.count (&BB) <= 0 && deleteBBs) 
       {
         dead.push_back (&BB);
         BB.dropAllReferences ();
         continue;
       }
-      
+
       if (BranchInst *br = dyn_cast<BranchInst> (BB.getTerminator ()))
       {
         if (br->isUnconditional ()) continue;
         BasicBlock* s0 = br->getSuccessor (0);
         BasicBlock* s1 = br->getSuccessor (1);
         
         BasicBlock* kill = NULL;
+     
         if (region.count (s0) <= 0)
           kill = s0;
         else if (region.count (s1) <= 0)
           kill = s1;
-        else continue;
-        
+        else 
+          continue;
+           
         Builder.SetInsertPoint (&BB, br);
         CallInst *ci = Builder.CreateCall
           (kill == s1 ? assumeFn : assumeNotFn, br->getCondition ());
         ci->setDebugLoc (br->getDebugLoc ());
         br->eraseFromParent ();
         Builder.SetInsertPoint (&BB);
         Builder.CreateBr (kill == s1 ? s0 : s1);
+        kill->removePredecessor(&BB);
```

Comment:
I think this has to be done before the branch instruction is deleted. The documentation says `Note that this should be called while the predecessor still refers to this block`. I think this means that `removePredecessor` is called first, then the branch is removed. Not sure it is necessary if we are deleting `kill` basic block anyways.